app/assets/javascripts/omni_nav/google-custom-search.js

```diff
@@ -0,0 +1,314 @@
+      $searchBox.find('input.gsc-search-button').on('click', onSearchEnter);
+      $searchBox.find('input.gsc-input').on('keyup', onSearchEnter);
+      $container.find('.search-filter-option').on('click', onSearchFilterClick);
+      $searchBox.find('.gsc-clear-button').on('click', hideResults);
+      $(window).on('resize', onSearchResultsResize);
+    };
+
+    var isOpen = function() {
+      return $container.hasClass('search-results-open');
+    };
+
+    var hideResults = function() {
+      // Reset results.
+      $searchResultsContainer.hide();
+      $omniNav.removeClass('search-results-open');
+      $container.removeClass('search-results-open');
+      unlockScroll();
+      gcsElement.clearAllResults();
+      $(document).off('keyup', onSearchEsc);
+      $(document).off('click', onDocumentClick);
+
+      // Clear search filters.
+      var hasSearchFilters = $container.find('.search-filter-option').length;
+      if ( hasSearchFilters ) $selectedSearchFilter.text(DEFAULT_FILTER_TEXT);
+    };
+
+    var renderGoogleSearchMarkup = function() {
+      // Each GCS element must have a unique id to use as its gname.
+      var searchBoxDiv = $searchBox[0];
+      var searchResultsDiv = $searchResults[0];
+
+      var searchBoxConfig = {
+        gname: containerId,
+        div: searchBoxDiv,
+        tag: 'searchbox',
+        attributes: {
+          enableAutoComplete: true,
+          autoCompleteMatchType: 'any',
+          resultSetSize: 6,
+          enableHistory: false
+        }
+      };
+
+      var searchResultsConfig = {
+        gname: containerId,
+        div: searchResultsDiv,
+        tag: 'searchresults',
+        attributes: {
+          linkTarget: '_self',
+          enableOrderBy: true
+        }
+      };
+
+      google.search.cse.element.render(searchBoxConfig, searchResultsConfig);
+    };
+
+    var updateSearchResultsHeight = function() {
+      $searchResultsContainer.height($(window).height());
+    };
+
+    var onSearchEnter = function(e) {
+      // The autocomplete container is not present until the user starts typing
+      // Add click event listener the first time it shows up
+      bindAutocomplete();
+
+      var resultsRequested = e.type == 'click' || e.which == ENTER_KEY;
+      var searchTermsPresent = gcsElement.getInputQuery().length > 0;
+      if( resultsRequested && searchTermsPresent ) {
+        showResults();
+      }
+    };
+
+    var onSearchFilterClick = function() {
+      $selectedSearchFilter.text($(this).text());
+    };
+
+    var onSearchResultsResize = function() {
+      clearTimeout(resizeTimeoutId);
+      resizeTimeoutId = setTimeout(updateSearchResultsHeight, 500);
+    };
+
+    var bindAutocomplete = function() {
+      // Autocomplete map. This seems very fragile as it requires Google to use
+      // the same hardcorded selector each time. But it is how this was originally coded
+      // and so far has proved reliable so I am keeping it for now.
```

Comment:
Yes, using the fragile selectors like this comment mentions is something we had to add when we started using two search boxes on the same page. If we only had one, we could simply do `$(".gsc-completion-container").on(...)` but now we have to figure out which is being clicked and the `gstl_50/51` was the one and only differentiating characteristic.

Reply:
It would be nice if the [google.search.cse.element object](https://developers.google.com/custom-search/docs/element#cse-element), `gcsElement`, offered access to the autocomplete element. That would make the most sense.